src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -1289,15 +1296,23 @@ void ObjectSynchronizer::inflate_helper(oop obj) {
   (void)inflate(Thread::current(), obj, inflate_cause_vm_internal);
 }
 
-// Can be called from non JavaThreads (e.g., VMThread) for FastHashCode
-// calculations as part of JVM/TI tagging.
-static bool is_lock_owned(Thread* thread, oop obj) {
-  assert(LockingMode == LM_LIGHTWEIGHT, "only call this with new lightweight locking enabled");
-  return thread->is_Java_thread() ? JavaThread::cast(thread)->lock_stack().contains(obj) : false;
+ObjectMonitor* ObjectSynchronizer::inflate(Thread* current, oop obj, const InflateCause cause) {
+  assert(current == Thread::current(), "must be");
+  if (LockingMode == LM_LIGHTWEIGHT && current->is_Java_thread()) {
+    return inflate_for(JavaThread::cast(current), obj, cause);
+  }
```

Comment:
Not quite what I had in mind. I expected the deopt code to call `inflate_for` to make it clear it was a different kind of inflation request.

Reply:
I think you are correct that it is more clear that way. Will combine this feedback with @coleenp feedback about have the inflate_impl just take the JavaThread* for LM_LIGHTWEIGHT.

Just need to figure out a parameter name. Might just call it `thread` and add a comment explaining it is something only used by the LM_LIGHTWEIGHT lockingmode.